crates/aptos-rest-client/src/lib.rs

```diff
@@ -510,7 +511,7 @@ impl Client {
         let response = self.inner.get(url.clone()).send().await?;
 
         if !response.status().is_success() {
-            let error_response = response.json::<AptosError>().await?;
+            let error_response = AptosError::parse_from_json(Some(response.json().await?));
```

Comment:
So standard serde doesn't work with the AptosError why?

Reply:
It should now that I added that snake case serde annotation to the enum, but this is safer, it's using the same serializer to deserialize that was originally used to serialize it. See https://github.com/aptos-labs/aptos-core/issues/3014 for more.